tests/README.md

```diff
@@ -0,0 +1,136 @@
+
+Inside the directory you must add a `CMakeLists.txt` file. You may use this template below. Please change the
+"your service" values to your actual service name.
+
+```
+# Copyright (c) 2020 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+cmake_minimum_required(VERSION 3.0.2)
+
+# change your-service to the name of your service
+set(TEST_NAME ble-service-your-service-unittest)
+
+add_executable(${TEST_NAME})
+
+target_include_directories(${TEST_NAME}
+    PRIVATE
+        .
+        # change the path to where your includes are
+        ${SERVICES_PATH}/YourService/include
+)
+
+target_sources(${TEST_NAME}
+    PRIVATE
+        # replace with the source file you have your unit test in
+        test_YourService.cpp
+        # replace the source file with the one that compiles your service
+        ${SERVICES_PATH}/YourService/source/YourService.cpp
+)
+
+target_link_libraries(${TEST_NAME}
+    PRIVATE
+        # add any stubs, headers or mocks you need here
+        # you will most likely want to include:
+        # mbed-os-fakes-ble
+        gmock_main
+)
+
+add_test(NAME "${TEST_NAME}" COMMAND ${TEST_NAME})
+```
+
+Inside CMake files for unit tests there are two special variables you can use that are set by the top level
+unit test CMakeLists.txt: `MBED_PATH`, `SERVICES_PATH`. These point to mbed-os and the services directory.
+
+## Stubs, mocks and fakes in mbed-os
+
+To make it easier to test your service mbed-os provides some ready made CMake libraries you can add in your
+`target_link_libraries` section (see template above). Libraries can be object files that contains stubs, mocks
+and fakes but also sets of include paths - this way you can gain access to normal headers from mbed-os.
+
+All the libraries available are in the `CMakeLists.txt` files in `mbed-os/UNITTESTS/stubs` and
+`mbed-os/UNITTESTS/fakes`.
+
+The most common ones you might need are `mbed-os-fakes-ble` and `mbed-os-fakes-event-queue`.
+
+### mbed-os-fakes-ble
+
+This library provides a fake BLE implementation that uses mocks instead of real BLE components for `Gap`,
+`GattServer`, `GattClient`, `SecurityManager`.
+
+There is no need to initialise a fake BLE instance; it is ready to go and can be used just like a normal BLE instance:
+```
+BLE ble = &BLE::Instance();
+```
+
+You can retrieve all the BLE APIs from the instance just like with a normal one:
+
+```
+Gap &gap = ble->gap();
+GattClient &client = ble->gattClient();
+GattServer &server = ble->gattServer();
+SecurityManager &sm = ble->securityManager();
+```
+
+But these are instead google mocks. This means you can set expectations on them. 
+
+```
+EXPECT_CALL(ble::gap_mock(), reset());
+```
+
+This will set up an expectations that at some point during the test the Gap::reset method will be called.
+
+The way google test works means that if you set any expectations on your mocks they must be destroyed at
```

Comment:
Expectations are destroyed when a Mock is destroyed, in the case of ble a singleton is used which makes things more difficult. I would advise on having `ble::reset_mocks` in the teardown of the fixture rather than at the end of each test.